docs-src/typescript/how-to-use-worker-versioning-in-typescript.md

```diff
@@ -0,0 +1,83 @@
+---
+id: how-to-use-worker-versioning-in-typescript
+title: How to use Worker Versioning in TypeScript
+sidebar_label: Worker Versioning
+description: Version your TypeScript Workers by using build ID-based versioning
+tags:
+  - typescript
+  - how-to
+---
+
+To use [Worker Versioning](/workers#worker-versioning) in TypeScript, you need to do the following:
+
+1. Determine and assign a Build ID to your built Worker code, and opt in to versioning.
+2. Tell the Task Queue your Worker is listening on about that Build ID, and whether its compatible with an existing Build ID.
+
+### Assign a Build ID to your Worker
+
+Let's say you've chosen `deadbeef` as your Build ID, which might be a short git commit hash (a reasonable choice as Build ID).
```

Comment:
raises the q in my mind of how do you get the commit hash of the commit that this code is in 😄 or is the hash always the last commit before the buildId is updated & code is deployed? wonder if that pattern is worth noting

Reply:
Yeah, it's a bit of a chicken-egg thing. You could save the hash into a file, and load that file when setting the Build ID.